### DISCUSSION TITLE
Hold charge between cycles

### DISCUSSION THREAD

**Discussion by alfwro13** [springfall2008/batpred#922] [Q&A]
I have recently moved to predbat and I'm quite happy with the way it functions.
However - I'm on Octopus Agile tariff for import and Agile fixed @15p/kwh export and with yesterday/today/tomorrow import prices being on 0 or negative half of the time this effectively means that predbat is doing full charge followed by full discharge - that is all good and makes perfect sense. 
One issue with that is that that after several none stop cycles (full charge and right after that full discharge, and repeat) both the battery and the inverter temperatures are quite high and for nearly 24h. The temperatures are still within the parameters but I was wondering if there is a way of telling predbat to hold the charge for 15-30 minutes between cycles - so charge fully, give the battery/inverter a rest, and discharge, and again give the system a rest the repeat the cycle ?
Is that possible? (I know I can manually force idle but was wondering if there is an automatic way of including that in the plan).
Thanks

**Comment by RobinCu** [comment]
You can set up an HA automation using the inverter or battery temperature as a fixed-number trigger (I use inverter temperature as a trigger to control a smart plug that powers a cooling fan). The action can be set to whatever you want, so it could be to switch on Predbat read-only mode, charge only mode, etc., and another to switch it back when a temperature drops below a fixed number. Let me know if I can help with some sample yaml.

**Comment by gcoan** [comment]
I too have automations to raise alarms when my battery or inverter temperatures are too high and next step will be to turn on fans on the inverter when it reaches a threshold.  Summer job!

All depends what you set the thresholds to of course but I find that my battery and inverter temperatures drop quite quickly after charging or discharging has finished, e.g.:
![IMG_0860](https://github.com/springfall2008/batpred/assets/142018870/0f8717b1-69e1-444c-a101-ff6a6e280fb3)

So if @alfwro13 your temperatures are staying high for 24 hours then that suggests something more significant with where your inverters/battery are placed and what cooling they are getting?
